src/frontend/src/instance/jaeger.rs

```diff
@@ -0,0 +1,336 @@
+        let selects = vec![
+            col(TRACE_ID_COLUMN),
+            col(TIMESTAMP_COLUMN),
+            col(DURATION_NANO_COLUMN),
+            col(SERVICE_NAME_COLUMN),
+            col(SPAN_NAME_COLUMN),
+            col(SPAN_ID_COLUMN),
+            col(SPAN_ATTRIBUTES_COLUMN),
+        ];
+
+        let filters = vec![col(TRACE_ID_COLUMN).eq(lit(trace_id))];
+
+        Ok(query_trace_table(
+            ctx,
+            self.catalog_manager(),
+            self.query_engine(),
+            selects,
+            filters,
+            Some(DEFAULT_LIMIT),
+            None,
+            false,
+        )
+        .await?)
+    }
+
+    async fn find_traces(
+        &self,
+        ctx: QueryContextRef,
+        query_params: QueryTraceParams,
+    ) -> ServerResult<Output> {
+        let selects = vec![
+            col(TRACE_ID_COLUMN),
+            col(TIMESTAMP_COLUMN),
+            col(DURATION_NANO_COLUMN),
+            col(SERVICE_NAME_COLUMN),
+            col(SPAN_NAME_COLUMN),
+            col(SPAN_ID_COLUMN),
+            col(SPAN_ATTRIBUTES_COLUMN),
+        ];
+
+        let mut filters = vec![];
+
+        if let Some(operation_name) = query_params.operation_name {
+            filters.push(col(SPAN_NAME_COLUMN).eq(lit(operation_name)));
+        }
+
+        if let Some(start_time) = query_params.start_time {
+            filters.push(col(TIMESTAMP_COLUMN).gt_eq(lit_timestamp_nano(start_time)));
+        }
+
+        if let Some(end_time) = query_params.end_time {
+            filters.push(col(TIMESTAMP_COLUMN).lt_eq(lit_timestamp_nano(end_time)));
+        }
+
+        if let Some(min_duration) = query_params.min_duration {
+            filters.push(col(DURATION_NANO_COLUMN).gt_eq(lit(min_duration)));
+        }
+
+        if let Some(max_duration) = query_params.max_duration {
+            filters.push(col(DURATION_NANO_COLUMN).lt_eq(lit(max_duration)));
+        }
+
+        Ok(query_trace_table(
+            ctx,
+            self.catalog_manager(),
+            self.query_engine(),
+            selects,
+            filters,
+            Some(DEFAULT_LIMIT),
+            query_params.tags,
+            false,
+        )
+        .await?)
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn query_trace_table(
+    ctx: QueryContextRef,
+    catalog_manager: &CatalogManagerRef,
+    query_engine: &QueryEngineRef,
+    selects: Vec<Expr>,
+    filters: Vec<Expr>,
+    limit: Option<usize>,
+    tags: Option<HashMap<String, JsonValue>>,
+    distinct: bool,
+) -> ServerResult<Output> {
+    let db = ctx.get_db_string();
+    let table = catalog_manager
+        .table(ctx.current_catalog(), &db, TRACE_TABLE_NAME, Some(&ctx))
+        .await
+        .context(CatalogSnafu)?
+        .with_context(|| TableNotFoundSnafu {
+            table: TRACE_TABLE_NAME,
+            catalog: ctx.current_catalog().to_string(),
+            schema: db.to_string(),
```

Comment:
Snafu will auto convert the string references into strings.
```suggestion
            catalog: ctx.current_catalog(),
            schema: db,
```